### PR TITLE
Errors in resize thumbnail

### PR DESCRIPTION
**Pull request by qdraw**
# PR Details 
## Description / Motivation and Context

This pull request introduces enhancements and fixes to the thumbnail generation functionality, improves error handling, and updates related tests. Key changes include better logging for failed operations, refactoring methods for clarity, and adding verification tests for specific scenarios.

### Thumbnail Generation Enhancements:

* Improved error logging in `ResizeThumbnailFromThumbnailImageHelper` to include more detailed information such as input thumbnail size and file hash with extension. (`starsky/starsky.foundation.thumbnailgeneration/GenerationFactory/ImageSharp/ResizeThumbnailFromThumbnailImageHelper.cs`, [starsky/starsky.foundation.thumbnailgeneration/GenerationFactory/ImageSharp/ResizeThumbnailFromThumbnailImageHelper.csL89-R90](diffhunk://#diff-f6bb93ce68591026f1a0119183fe031435c8b25c82aee2e76b1b35247ffe89ebL89-R90))
* Added a helper method `UpdateAllSizesToFailure` in `SharedGenerate` to handle failure scenarios for all thumbnail sizes, ensuring consistent error reporting. (`starsky/starsky.foundation.thumbnailgeneration/GenerationFactory/Shared/SharedGenerate.cs`, [starsky/starsky.foundation.thumbnailgeneration/GenerationFactory/Shared/SharedGenerate.csL48-R102](diffhunk://#diff-0f727249ea29c0f9b3aaaf19db209d634361bff110e46cdc102c77711de53a2dL48-R102))

### Code Refactoring:

* Renamed delegate parameter `resizeDelegate` to `resizeThumbnailFromSourceImage` for better readability and consistency. (`starsky/starsky.foundation.thumbnailgeneration/GenerationFactory/Shared/SharedGenerate.cs`, [starsky/starsky.foundation.thumbnailgeneration/GenerationFactory/Shared/SharedGenerate.csL28-R29](diffhunk://#diff-0f727249ea29c0f9b3aaaf19db209d634361bff110e46cdc102c77711de53a2dL28-R29))
* Added missing `using` statements for necessary namespaces in test files. (`starsky/starsky.foundation.thumbnailgeneration/GenerationFactory/Shared/SharedGenerate.cs`, [[1]](diffhunk://#diff-0f727249ea29c0f9b3aaaf19db209d634361bff110e46cdc102c77711de53a2dR8); `starsky/starskytest/starsky.foundation.thumbnailgeneration/GenerationFactory/ThumbnailServiceTests.cs`, [[2]](diffhunk://#diff-d4e526b0fbe546b1c1e0662aec5805e32efc2df3e4b102d5ba47674ef70b4e9cR15-R25)

### Test Improvements:

* Introduced verification tests to validate the output of thumbnail generation in failure scenarios, including cases where the image type is incorrect or the file is corrupt. (`starsky/starskytest/starsky.foundation.thumbnailgeneration/GenerationFactory/ThumbnailServiceTests.cs`, [[1]](diffhunk://#diff-d4e526b0fbe546b1c1e0662aec5805e32efc2df3e4b102d5ba47674ef70b4e9cR120-R135) [[2]](diffhunk://#diff-d4e526b0fbe546b1c1e0662aec5805e32efc2df3e4b102d5ba47674ef70b4e9cR249-R263)
* Added new verified test files to ensure consistent output for failure cases. (`starsky/starskytest/starsky.foundation.thumbnailgeneration/GenerationFactory/ThumbnailServiceTests.GenerateThumbnail_FileHash_WrongImageType_Verify.verified.txt`, [[1]](diffhunk://#diff-17231a6aeb0ce8c5b6f9aaf1ab35063c6dec4d85f89db88cd4f57e769c27176eR1-R42); `starsky/starskytest/starsky.foundation.thumbnailgeneration/GenerationFactory/ThumbnailServiceTests.GenerateThumbnail__Corrupt_Verify.verified.txt`, [[2]](diffhunk://#diff-e1a7d947e385685c5431fa890ece002c1abce03bc5738c68c823782684a0e42aR1-R46)

These changes collectively improve the robustness of the thumbnail generation process and ensure better test coverage for edge cases.

## How Has This Been Tested

- [ ] C# Unit tests 
- [ ] Typescript Unit tests 
- [ ] Other Unit tests
- [ ] Manual tests
- [ ] Automatic End2end tests (starsky-tools/end2end)

## Types of changes

- [ ] Added _for new features_
- [ ] Breaking change _fix or feature that would cause existing functionality to change_
- [ ] Changed _for non-breaking changes in existing functionality for example docs change / refactoring / dependency upgrades_
- [ ] Deprecated _for soon-to-be removed features_
- [ ] Removed _for now removed features_
- [ ] Fixed _for any bug fixes_
- [ ] Security _in case of vulnerabilities_

## Checklist

- [ ] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly (update when needed)
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have updated the **./history.md** document

### CODE REVIEW

File: starsky/starsky.foundation.thumbnailgeneration/GenerationFactory/Shared/SharedGenerate.cs [hunk 1]
```diff
@@ -45,14 +45,22 @@ public async Task<IEnumerable<GenerationResultModel>> GenerateThumbnail(
...
-				imageFormat);
+		var largeImageResult = await resizeThumbnailFromSourceImage(toGenerateSize,
+			singleSubPath, fileHash, imageFormat);
+
+		if ( !largeImageResult.Success )
+		{
+			logger.LogError(
+				$"[SharedGenerate] ResizeThumbnailFromSourceImage failed for " +
+				$"S: {singleSubPath} - H: {fileHash} SI: {toGenerateSize}");
+			return preflightResult.AddOrUpdateRange([largeImageResult]);
```
Comment (qdraw) [inline starsky/starsky.foundation.thumbnailgeneration/GenerationFactory/Shared/SharedGenerate.cs:56]: need to check the output array
